src/main/java/redis/clients/jedis/JedisSocketFactory.java

```diff
@@ -15,23 +16,29 @@
  */
 public interface JedisSocketFactory {
```

Comment:
@sazzad16 it seems like all the methods are deprecated so how come the interface isn't?

Reply:
@gkorland `createSocket()` method is here to stay. But instead of throwing IOException, it will throw only JedisConnectionException.

Reply:
Yes, since you are not going to really deprecate it